clang/lib/StaticAnalyzer/Checkers/MallocChecker.cpp

```diff
@@ -194,7 +226,7 @@ class RefState {
   void Profile(llvm::FoldingSetNodeID &ID) const {
     ID.AddInteger(K);
     ID.AddPointer(S);
-    ID.AddInteger(Family);
+    ID.AddInteger(Family.kind());
```

Comment:
This Profile wouldn't take the custom allocation function name into account.
I'd suggest letting the Family define its own Profile method and just pass the ID to it.
That is the canonical way of doing this.